src/librustc_mir/interpret/step.rs

```diff
@@ -240,6 +240,23 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
 
             Ref(_, _, ref place) => {
+                // FIXME(wesleywiser) we don't currently handle the case where we try to make a ref
```

Comment:
What does "not handling" mean? Does it ICE the compiler? Does it create weird states in const prop that we don't want?

Reply:
This test case:

https://github.com/rust-lang/rust/blob/f43ac065347ccbd80d0465489f51408fd907bcf7/src/test/ui/unsized-locals/autoderef.rs#L10

ICE's the compiler at this expect:

https://github.com/rust-lang/rust/blob/f43ac065347ccbd80d0465489f51408fd907bcf7/src/librustc_mir/interpret/eval_context.rs#L461

Reply:
I can't come up with a good way to resolve this, but if it's solely arguments, maybe check in const prop whether any statement mentions an argument-local and skip that statement entirely in that case.

Reply:
Ok, I'll look into that. I wasn't sure if this was an appropriate place to fix that bug anyway but it seemed to work.

Reply:
Resolved in cfd0efd5a917556d9410c0fa1f3637e6ee51df68